src/sql/src/plan/decorrelate.rs

```diff
@@ -417,14 +417,87 @@ impl ScalarExpr {
                     .collect::<Vec<_>>(),
             },
             If { cond, then, els } => {
-                // TODO(jamii) would be nice to only run subqueries in `then` when `cond` is true
-                // (if subqueries later gain the ability to throw errors, this impacts correctness too)
-                // NOTE: This also affects performance *now* if either branch is expensive to produce
-                // and/or maintain but is not the returned result.
-                SS::If {
-                    cond: Box::new(cond.applied_to(id_gen, col_map, inner)),
-                    then: Box::new(then.applied_to(id_gen, col_map, inner)),
-                    els: Box::new(els.applied_to(id_gen, col_map, inner)),
+                // If is complicated by the fact that we do not want to determine
+                // the `then` or `else` branch for tuples that do not do not pass
+                // the `cond` test. Our strategy is to first determine cond, and
```

Comment:
I think this sentence is missing its

Reply:
This text has many problems, and shall be replaced!